packages/dumb/src/components/stats/insulin/insulin-stat.tsx

```diff
@@ -0,0 +1,200 @@
+ *    this list of conditions and the following disclaimer in the documentation
+ *    and/or other materials provided with the distribution.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+ * AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+ * IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE
+ * DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE LIABLE
+ * FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL
+ * DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR
+ * SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
+ * CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY,
+ * OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+ * OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+import React, { type FunctionComponent, memo, useMemo } from 'react'
+import styles from './insulin-stat.css'
+import Box from '@mui/material/Box'
+import Chip from '@mui/material/Chip'
+import { t } from 'i18next'
+import { formatDecimalNumber } from '../../../utils/format/format.util'
+import { StatTooltip } from '../../tooltips/stat-tooltip/stat-tooltip'
+import { useLocation } from 'react-router-dom'
+import { EMPTY_DATA_PLACEHOLDER } from '../../../models/stats.model'
+
+interface TotalInsulinPropsData {
+  id: string
+  title: string
+  units: string
+  value: number
+  valueString: string
+}
+
+export interface TotalInsulinStatProps {
+  data: TotalInsulinPropsData[]
+  totalInsulin: number
+  weight: number
+  dailyDose: number
+}
+
+const InsulinStat: FunctionComponent<TotalInsulinStatProps> = (props) => {
+  const {
+    data,
+    totalInsulin,
+    dailyDose,
+    weight
+  } = props
+  const location = useLocation()
+  const isDailyPage = location.pathname.includes('daily')
+  const isDashboardPage = location.pathname.includes('dashboard')
+  const isTrendsPage = location.pathname.includes('trends')
+  const annotations = [t(isDailyPage ? 'total-insulin-days-tooltip' : 'average-daily-insulin-tooltip'), t('total-insulin-how-calculate-tooltip')]
+  const isEmptyWeight = weight === 0
+  const weightValue = isEmptyWeight ? EMPTY_DATA_PLACEHOLDER : weight
+
+  if (data.length === 0) {
+    annotations.push(t('tooltip-empty-stat'))
+  }
+
+  const dailyDosePerWeight = useMemo(() => {
+    if (isEmptyWeight) {
+      return EMPTY_DATA_PLACEHOLDER
+    }
+    const value = dailyDose / weight
+    return value > 0 && Number.isFinite(value) ? formatDecimalNumber(value, 2) : EMPTY_DATA_PLACEHOLDER
+  }, [dailyDose, isEmptyWeight, weight])
+
+  const outputValueClasses = (): string => {
+    const isDisabled = dailyDosePerWeight === EMPTY_DATA_PLACEHOLDER
+    return `${isDisabled ? `${styles.outputValueDisabled}` : `${styles.dailyDoseValue}`}`
+  }
+
+  const percent = (value: number): string => {
```

Comment:
That's not a good function name, must represent an action or something that the function is doing. I suggest `getPercentage`